Framework/Foundation/include/Framework/Pack.h

```diff
@@ -248,6 +257,25 @@ constexpr auto concatenate_pack_unique(P1 p1, P2 p2, Ps... ps)
 template <typename... Ps>
 using concatenated_pack_unique_t = decltype(concatenate_pack_unique(Ps{}...));
 
+template <typename T, typename T2>
+struct add_type_impl;
+template <typename T, typename... Ts>
+struct add_type_impl<T, pack<Ts...>> {
+  using type = std::conditional_t<sizeof...(Ts) == 0, pack<T>, std::conditional_t<has_type_v<T, pack<Ts...>>, pack<Ts...>, pack<T, Ts...>>>;
+};
+template <typename T, typename T2>
+using add_type = typename add_type_impl<T, T2>::type;
+
+template <typename P, typename PT = pack<>>
+struct unique_types_impl {
+  using type = PT;
+};
+template <typename PT, typename T, typename... Ts>
+struct unique_types_impl<pack<T, Ts...>, PT> : unique_types_impl<pack<Ts...>, add_type<T, PT>> {
+};
+template <typename P>
+using unique_types = typename unique_types_impl<P>::type;
+
```

Comment:
You should try to follow the conventions which we use in the rest of the file. In particular:

* `unique_types` /  `add_type` should probably be `uniq_pack_t`
* `add_type` should probably be renamed to`set_insert_pack` or something like that.

I also wonder if the whole logic would not be better expressed by a constexpr function, like done elsewhere.

Reply:
Ok, indeed it is possible